From 8ba366367f7c7e4a92e833030edb4aef9e01f51c Mon Sep 17 00:00:00 2001
From: Wei Yongjun <weiyongjun1@huawei.com>
Date: Fri, 7 Jan 2022 23:07:14 +0800
Subject: [PATCH] tcp_comp: add Kconfig for tcp payload compression
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 8ba366367f7c7e4a92e833030edb4aef9e01f51c


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4PNEK
CVE: NA

-------------------------------------------------

Add config item CONFIG_TCP_COMP for tcp payload compression.

This allows payload compression handling of the TCP protocol
to be done in-kernel.

This patch only adds the CONFIG_TCP_COMP config, tcp compression
capability is implemented later.

Signed-off-by: Wei Yongjun <weiyongjun1@huawei.com>
Signed-off-by: Wang Yufen <wangyufen@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Lu Wei <luwei32@huawei.com>
Reviewed-by: Wei Yongjun <weiyongjun1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 net/ipv4/Kconfig | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/net/ipv4/Kconfig b/net/ipv4/Kconfig
index 87983e70f03f..59ffbf80f7f2 100644
--- a/net/ipv4/Kconfig
+++ b/net/ipv4/Kconfig
@@ -742,3 +742,11 @@ config TCP_MD5SIG
 	  on the Internet.
 
 	  If unsure, say N.
+
+config TCP_COMP
+	bool "TCP: Transport Layer Compression support"
+	help
+          Enable kernel payload compression support for TCP protocol. This allows
+          payload compression handling of the TCP protocol to be done in-kernel.
+
+	  If unsure, say Y.
-- 
2.26.2

